pkg/apiserver/rest/asset_scan_estimation_controller.go

```diff
@@ -0,0 +1,151 @@
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return sendError(ctx, http.StatusNotFound, fmt.Sprintf("asset scan estimation was not found. assetScanEstimationID=%v: %v", assetScanEstimationID, err))
+		}
+		return sendError(ctx, http.StatusInternalServerError, fmt.Sprintf("failed to get asset scan estimation. assetScanEstimationID=%v: %v", assetScanEstimationID, err))
+	}
+
+	// PATCH request might not contain the ID in the body, so set it from
+	// the URL field so that the DB layer knows which object is being updated.
+	if assetScanEstimation.Id != nil && *assetScanEstimation.Id != assetScanEstimationID {
+		return sendError(ctx, http.StatusBadRequest, fmt.Sprintf("id in body %s does not match object %s to be updated", *assetScanEstimation.Id, assetScanEstimationID))
+	}
+	assetScanEstimation.Id = &assetScanEstimationID
+
+	updatedAssetScanEstimation, err := s.dbHandler.AssetScanEstimationsTable().UpdateAssetScanEstimation(assetScanEstimation, params)
+	if err != nil {
+		var validationErr *common.BadRequestError
+		var conflictErr *common.ConflictError
+		var preconditionFailedErr *databaseTypes.PreconditionFailedError
+		switch true {
+		case errors.As(err, &conflictErr):
+			existResponse := &models.AssetScanEstimationExists{
+				Message:             utils.PointerTo(conflictErr.Reason),
+				AssetScanEstimation: &updatedAssetScanEstimation,
+			}
+			return sendResponse(ctx, http.StatusConflict, existResponse)
+		case errors.As(err, &validationErr):
+			return sendError(ctx, http.StatusBadRequest, err.Error())
+		case errors.As(err, &preconditionFailedErr):
+			return sendError(ctx, http.StatusPreconditionFailed, err.Error())
+		default:
+			return sendError(ctx, http.StatusInternalServerError, fmt.Sprintf("failed to update asset scan estimation in db. assetScanEstimationID=%v: %v", assetScanEstimationID, err))
+		}
+	}
+
+	return sendResponse(ctx, http.StatusOK, updatedAssetScanEstimation)
+}
+func (s *ServerImpl) PutAssetScanEstimationsAssetScanEstimationID(ctx echo.Context, assetScanEstimationID models.AssetScanEstimationID, params models.PutAssetScanEstimationsAssetScanEstimationIDParams) error {
+	var assetScanEstimation models.AssetScanEstimation
+	err := ctx.Bind(&assetScanEstimation)
+	if err != nil {
+		return sendError(ctx, http.StatusBadRequest, fmt.Sprintf("failed to bind request: %v", err))
+	}
+
+	// check that an asset scan estimation with that id exists.
+	_, err = s.dbHandler.AssetScanEstimationsTable().GetAssetScanEstimation(assetScanEstimationID, models.GetAssetScanEstimationsAssetScanEstimationIDParams{})
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return sendError(ctx, http.StatusNotFound, fmt.Sprintf("asset scan estimation was not found. assetScanEstimationID=%v: %v", assetScanEstimationID, err))
+		}
+		return sendError(ctx, http.StatusInternalServerError, fmt.Sprintf("failed to get asset scan estimation. assetScanEstimationID=%v: %v", assetScanEstimationID, err))
+	}
```

Comment:
Ditto, just need to move the ErrRecordNotFound error handling to the block below here.